src/components/Carousel/Carousel.js

```diff
@@ -0,0 +1,258 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import { Button, ButtonIcon } from '../../';
+
+class Carousel extends Component {
+  state = {
+    activeIndex: 0,
+    panels: [],
+  }
+
+  autoPlayId = null // eslint-disable-line react/sort-comp
+  autoPlayFlavors = ['icon-border-filled', 'icon-x-small'];
+
+  componentWillMount() {
+    const { autoPlayActive, children } = this.props;
+
+    this.updatePanels(children);
+
+    if (autoPlayActive) {
+      this.startAutoPlay();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.autoPlayId) {
+      clearInterval(this.autoPlayId);
+    }
+  }
+
+  componentWillReceiveProps({ autoPlayActive: nextAutoPlayActive, children: nextChildren }) {
+    const { children: prevChildren } = this.props;
+    const { autoPlayActive: prevAutoPlayActive } = this.state;
+
+    if (nextChildren !== prevChildren) {
+      this.updatePanels(nextChildren);
+    }
+
+    if (!!nextAutoPlayActive !== prevAutoPlayActive) {
+      if (nextAutoPlayActive) this.startAutoPlay();
+      else this.stopAutoPlay();
+    }
+  }
+
+  onClickIndicator = e => this.setActivePanel(Number(e.target.dataset.index));
```

Comment:
This should also stop or reset the autoplay when active. currently, there is an edge-case where two indicators appear to be active when autoplay is active and an item is clicked, see attached screenshot.

<img width="871" alt="screen shot 2018-01-25 at 17 05 12" src="https://user-images.githubusercontent.com/1682504/35398446-2d273746-01f2-11e8-87a6-349cb18cfa2c.png">

Reply:
This was actually an issue with the indicator keeping its `:focus` state. I now added a blur() to the click event, not sure though whether this is the best solution ...